### DISCUSSION TITLE
Change module title dynamically

### DISCUSSION THREAD

**Discussion by tiggertag** [oqtane/oqtane.framework#2350] [Q&A] [answered]
This might be a simple thing, but how do I change the module title dynamically at runtime?

**Comment by sbwalker** [comment]
Currently the framework does not support this capability. The INotifyPropertyChanged feature introduced in Oqtane 3.1.0 should make this easier to implement (and would provide a good core example of how to use the feature).

**Comment by sbwalker** [comment, marked as answer]
@tiggertag #2358 provides the feature you were looking for. The methods were added to ModuleBase so in your module razor components you simply need to call:

SetModuleTitle("Your Custom Title");

I also added a feature which allows the module visibility to be specified programmatically as well:

SetModuleVisibility(false);

(this would be used in scenarios where you want to show/hide a module based on some custom logic - and it would hide the module's container as well )

**Reply by tiggertag** [reply]
Thank you! Will try it.
